test/integration/concierge_webhookauthenticator_status_test.go

```diff
@@ -0,0 +1,296 @@
+							{
+								Type:    "Ready",
+								Status:  "False",
+								Reason:  "NotReady",
+								Message: "the WebhookAuthenticator is not ready: see other conditions for details",
+							}, {
+								Type:    "AuthenticatorValid",
+								Status:  "Unknown",
+								Reason:  "UnableToValidate",
+								Message: "unable to validate; see other conditions for details",
+							}, {
+								Type:    "TLSConnectionNegotiationValid",
+								Status:  "False",
+								Reason:  "UnableToDialServer",
+								Message: "cannot dial server: dial tcp 127.0.0.1:443: connect: connection refused",
+							},
+						},
+					))
+			},
+		},
+	}
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.run(t)
+		})
+	}
+}
+
+func TestConciergeWebhookAuthenticatorCRDValidations_Parallel(t *testing.T) {
+	env := testlib.IntegrationEnv(t)
+	webhookAuthenticatorClient := testlib.NewConciergeClientset(t).AuthenticationV1alpha1().WebhookAuthenticators()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	t.Cleanup(cancel)
+
+	objectMeta := testlib.ObjectMetaWithRandomName(t, "webhook-authenticator")
+	tests := []struct {
+		name                 string
+		webhookAuthenticator *v1alpha1.WebhookAuthenticator
+		wantErr              string
+	}{
+		{
+			name: "endpoint can not be empty string",
+			webhookAuthenticator: &v1alpha1.WebhookAuthenticator{
+				ObjectMeta: objectMeta,
+				Spec: v1alpha1.WebhookAuthenticatorSpec{
+					Endpoint: "",
+				},
+			},
+			wantErr: `WebhookAuthenticator.authentication.concierge.` + env.APIGroupSuffix + ` "` + objectMeta.Name + `" is invalid: ` +
+				`spec.endpoint: Invalid value: "": spec.endpoint in body should be at least 1 chars long`,
+		},
+		{
+			name: "endpoint must be https",
+			webhookAuthenticator: &v1alpha1.WebhookAuthenticator{
+				ObjectMeta: objectMeta,
+				Spec: v1alpha1.WebhookAuthenticatorSpec{
+					Endpoint: "http://www.example.com",
+				},
+			},
+			wantErr: `WebhookAuthenticator.authentication.concierge.` + env.APIGroupSuffix + ` "` + objectMeta.Name + `" is invalid: ` +
+				`spec.endpoint: Invalid value: "http://www.example.com": spec.endpoint in body should match '^https://'`,
+		},
+		{
+			name: "minimum valid authenticator",
+			webhookAuthenticator: &v1alpha1.WebhookAuthenticator{
+				ObjectMeta: testlib.ObjectMetaWithRandomName(t, "webhook"),
+				Spec: v1alpha1.WebhookAuthenticatorSpec{
+					Endpoint: "https://localhost/webhook-isnt-actually-here",
+				},
+			},
+		},
+		{
+			name: "valid authenticator can have empty TLS block",
+			webhookAuthenticator: &v1alpha1.WebhookAuthenticator{
+				ObjectMeta: testlib.ObjectMetaWithRandomName(t, "webhook"),
+				Spec: v1alpha1.WebhookAuthenticatorSpec{
+					Endpoint: "https://localhost/webhook-isnt-actually-here",
+					TLS:      &v1alpha1.TLSSpec{},
+				},
+			},
+		},
+		{
+			name: "valid authenticator can have empty TLS CertificateAuthorityData",
```

Comment:
The name of this test does not match what it is testing. I suspect the name of the test is just a typo.

Reply:
Yup, will fix.